services/provider/client/client.go

```diff
@@ -24,17 +24,33 @@ func NewConsumerClient(cc *grpc.ClientConn, timeout time.Duration) *ConsumerClie
 func NewGRPCConnection(serverAddr string, opts []grpc.DialOption) (*grpc.ClientConn, error) {
 	conn, err := grpc.Dial(serverAddr, opts...)
 	if err != nil {
-		klog.Fatalf("failed to dial: %v", err)
+		return nil, fmt.Errorf("failed to dial: %v", err)
 	}
 	return conn, err
 }
 
+// Initialize initialzes a new provider client
+func Initialize(serverAddr, token string, reqTimeOut time.Duration) (*ConsumerClient, error) {
+	interceptor := NewAuthInterceptor(token)
+
+	// TODO: ADD TLS support
+	opts := []grpc.DialOption{
+		grpc.WithUnaryInterceptor(interceptor.Unary()),
+	}
+	cc, err := NewGRPCConnection(serverAddr, opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create gRPC connection: %v", err)
+	}
+
+	return NewConsumerClient(cc, reqTimeOut), nil
+}
+
 // OnBoardConsumer to validate the consumer and create StorageConsumer
 // resource on the StorageProvider cluster
-func (cc *ConsumerClient) OnBoardConsumer(token, capacity string) (*pb.OnBoardConsumerResponse, error) {
+func (cc *ConsumerClient) OnBoardConsumer(name, capacity string) (*pb.OnBoardConsumerResponse, error) {
 	req := &pb.OnBoardConsumerRequest{
-		Token:    token,
-		Capacity: capacity,
+		ConsumerName: name,
+		Capacity:     capacity,
```

Comment:
Probably move this to the first commit where there's related changes.

Reply:
Right. Moved it to the first commit where proto file was updated.